### PR TITLE
Adding overrides and enable override warning for GCC

### LINKED ISSUE #1826: Adding missing override warning
Following the discussion in #589
Since we recently added the missing overrides, we could think abt adding the override warnings to the global CMakeLists.

Basically what them DEM guys [are already doing](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/DEM_application/CMakeLists.txt#L109-L115) 

This might lead to many warnings with external libs (e.g. Trilinos gives maaany warnings), so it would be good if one could disable the check for certain apps (is this possible?)

@KratosMultiphysics/technical-committee

### PR DESCRIPTION
**Pull request by philbucher**
I added quite some missing overrides in the Apps that I am familiar with

plus I used the suggestions from #1826 to enable the missing-overrides warning in GCC
=> I found a way to ignore the warnings from Trilinos by adding a `SYSTEM` command in CMakeLists.txt (I did the same for the MPI-libs)
@roigcarlo could you please have a look if this is ok (its the 2. last commit)? Thx

This will create some warnings in the other apps that I am not familiar with but I think it is worth fixing it
If you do not agree we can also just merge the missing overrides

@maceligueta if this is ok for being merged you could remove the statement from your cmakelists since it will be enabled on a global level

closes #1826

### CODE REVIEW

**Comment by philbucher** [comment]
@maceligueta Travis showed now some override warnings that had not been there before as far as I remember. Maybe the addition of the warning was not working properly before?

I added the missing overrides and also removed the warning addition in your cmakelists, hope this was ok

**Comment by RiccardoRossi** [comment]
I am very much in favor of this change, however i would like the @KratosMultiphysics/technical-committee to approve (or to do it after discussing it). Adding it for tomorrow's meeting

**Comment by maceligueta** [comment]
@philbucher thanks. If it's approved tomorrow I am ok with removing it from the DEM app, sure.

File: CMakeLists.txt [hunk 1]
```diff
@@ -317,7 +322,7 @@ if(${MPI_NEEDED} MATCHES ON )
   set(CMAKE_CXX_LINK_FLAGS ${CMAKE_CXX_LINK_FLAGS} ${MPI_LINK_FLAGS})
 
   # Note: that this will be included in the compilation of the kratos core, so that it is not needed to include it somewhere else
-  include_directories(${MPI_INCLUDE_PATH})
+  include_directories(SYSTEM ${MPI_INCLUDE_PATH})
```
Comment (roigcarlo) [inline CMakeLists.txt:325]: My only concern is this. Is strictly necessary? I remember having problems with boost while trying to suppress warnings this way if there was already a system version of the library. For this case I am thinking mostly on clusters ( they should have MPI implementation in modules but you can never be 100% sure)

That said, I've tried in my computer and it works. I comment it anyway so if it gives problems we have a hint on this.

**Review by roigcarlo — APPROVED** [review]
:+1: Seems fine. Sorry for the delay reviewing this.
Just a comment but it's mostly for myself as a reminder in case I try to compiler in certain clusters

File: CMakeLists.txt (hunk 1 above)
Comment (philbucher) [reply CMakeLists.txt:325]: I get warnings abt overrides from MPI (openMPI 1.10 in my case) if I don't do this

What exactly were the problems? Shouldn't MPI always be seen as "System", independent if it is installed by yourself or provided?

Comment (philbucher) [reply CMakeLists.txt:325]: from CMake docu:

If the SYSTEM option is given, the compiler will be told the directories are meant as system include directories on some platforms. Signalling this setting might achieve effects such as the compiler skipping warnings, or these fixed-install system files not being considered in dependency calculations - see compiler docs.

Comment (roigcarlo) [reply CMakeLists.txt:325]: Yep is as the doc says. but as I said, no problem right now.

**Comment by philbucher** [comment]
If @KratosMultiphysics/technical-committee does not have further objections I will merge this on Wednesday

File: CMakeLists.txt (hunk 1 above)
Comment (philbucher) [reply CMakeLists.txt:325]: Ok thx

**Pull request merged**